pkg/trace/obfuscate/sql.go

```diff
@@ -165,6 +165,19 @@ func (f *groupingFilter) Reset() {
 // some elements such as comments and aliases and obfuscation attempts to hide sensitive information
 // in strings and numbers by redacting them.
 func (o *Obfuscator) ObfuscateSQLString(in string) (*ObfuscatedQuery, error) {
+	if v, ok := o.queryCache.Get(in); ok {
+		return v.(*ObfuscatedQuery), nil
+	}
+	oq, err := o.obfuscateSQLString(in)
+	if err != nil {
+		return oq, err
+	}
+	cost := len(in) + len(oq.Query) + len(oq.TablesCSV)
```

Comment:
Is this cost calculation is right? Ristretto doesn't store the key (`in`), but 2 64-bit hashes of the key.

Reply:
Then I guess it would be 2 * 64 bits = 16 bytes, instead of len(in) bytes? I’ll check this out tomorrow, thanks for looking deeply.

Reply:
I dropped the `len(in)` value completely and added a `Cost` method to `ObfuscatedQuery`. Let me know what you think now. I also fine-tuned the `*ristretto.Config` a bit knowing this, in my latest commit.

Reply:
👍